src/diff/props.js

```diff
@@ -116,6 +116,14 @@ export function setProperty(dom, name, value, oldValue, isSvg) {
  */
 function eventProxy(e) {
 	this._listeners[e.type + false](options.event ? options.event(e) : e);
+	if (this._isControlled && this._prevValue) {
```

Comment:
After something potentially altered the state we check if this is a controlled component with a previous value if so set the previous value